presto-spark-base/src/main/java/com/facebook/presto/spark/execution/BroadcastInfo.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * This class is a 1:1 strict API mapping to BroadcastInfo in
+ * presto-native-execution/presto_cpp/main/operators/BroadcastFactory.h.
+ * Please refrain changes to this API class. If any changes have to be made to
+ * this class, one should make sure to make corresponding changes in the above
+ * C++ struct and its corresponding serde functionalities.
+ */
+public class BroadcastInfo
```

Comment:
Assuming this class will not be needed if the split generation change I suggested in other comments is acceptable.

Reply:
Correct, removed